src/Microsoft.ML.Data/Transforms/ColumnConcatenatingEstimator.cs

```diff
@@ -109,6 +115,10 @@ private bool HasCategoricals(SchemaShape.Column col)
             return new SchemaShape.Column(name, vecKind, itemType, false, new SchemaShape(meta));
         }
 
+        /// <summary>
+        /// Returns the <see cref="SchemaShape"/> of the schema which will be produced by the transformer.
+        /// Used for schema propagation and verification in a pipeline.
+        /// </summary>
         public SchemaShape GetOutputSchema(SchemaShape inputSchema)
```

Comment:
>GetOutputSchema [](start = 27, length = 15)

Not a request and probably not related to you. Just feel if we could have documentations for how we compute `SchemaShape` in `GetOutputSchema`, we might have documented everything needed by users.

Reply:
I agree, although I think we should write that in the constructor, or the summary for the class itself. I think we should add it as part of another PR that focuses on improving doc, my PR is a larger set of changes focused on internalization and adding minimal documentation where it was not present.

---
In reply to: [253243462](https://github.com/dotnet/machinelearning/pull/2363#discussion_r253243462) [](ancestors = 253243462)